### PR TITLE
Updated bulletproofs-and-mimblewimble report

### PR DESCRIPTION
**Pull request by hansieodendaal**
Updated bulletproofs-and-mimblewimble report with compression of vector commitments

### CODE REVIEW

File: src/cryptography/bulletproofs-and-mimblewimble/MainReport.md [hunk 1]
```diff
@@ -40,16 +40,29 @@ The essence of Bulletproofs is its inner-product algorithm originally presented
...
+
+<p align="center"><img src="sources/VectorCutNHalf.png" width="450" /></p>
+
+<p align="center"><b>Figure&nbsp;1: Vector Pedersen Commitment Cut and Half ([<a href="https://eprint.iacr.org/2016/263.pdf" title="Efficient zero-knowledge arguments for arithmetic 
+circuits in the discrete log setting, Bootle J et al.">12</a>], [<a href="https://www.benthamsgaze.org/2016/10/25/how-to-do-zero-knowledge-from-discrete-logs-in-under-7kb/" title="How to do Zero-Knowledge from Discrete-Logs 
+in under 7kB, Bootle J.">63</a>])</b></p>
+
+
+
+Bulletproofs only rely on the discrete logarithm assumption. What this means in practice is that Bulletproofs are compatible with any secure elliptic curve, which makes it extremely versatile. The proof sizes are short; only $ [2 \log_2(n) + 9] $ elements are required for the range proofs and $ [\log_2(n) + 13] $ elements for arithmetic circuit proofs with $ n $ denoting the multiplicative complexity. The logarithmic proof size additionally enables the *prover* to aggregate multiple range proofs into a single short proof, as well as to aggregate multiple range proofs from different parties into one proof (see Figure&nbsp;1). ([[1]], [[3]], [[5]])
```
Comment (neonknight64) [inline src/cryptography/bulletproofs-and-mimblewimble/MainReport.md]: Consider changing "The logarithmic proof size additionally enables" to "Additionally, the logarithmic proof size enables".

File: src/cryptography/bulletproofs-and-mimblewimble/MainReport.md [hunk 2]
```diff
@@ -40,16 +40,29 @@ The essence of Bulletproofs is its inner-product algorithm originally presented
...
 
 <p align="center"><img src="sources/AggregateBulletproofsSize.png" width="650" /></p>
 
 <p align="center"><b>Figure&nbsp;1: Logarithmic Aggregate Bulletproofs Proof Sizes [<a href="https://drive.google.com/file/d/18OTVGX7COgvnZ7T0ke-ajhMWwOHOWfKV/view" title="Bullet Proofs (Slides), 
 Bitcoin Milan Meetup 2018-02-02, 
 Andrew Poelstra">3</a>]</b></p>
 
+
 If all Bitcoin transactions were confidential, approximately 50 million UTXOs from approximately 22 million transactions would result in roughly 160GB of range proof data, when using current/linear proof systems and assuming use of 52-bits to represent any value from 1 satoshi up to 21 million bitcoins. Aggregated Bulletproofs would reduce that data set to less than 17GB. [[1]]
```
Comment (neonknight64) [inline src/cryptography/bulletproofs-and-mimblewimble/MainReport.md]: Consider changing "that data set" to "the data storage requirement".

File: src/cryptography/bulletproofs-and-mimblewimble/MainReport.md [hunk 3]
```diff
@@ -40,16 +40,29 @@ The essence of Bulletproofs is its inner-product algorithm originally presented
...
+The basis of confidential transactions is to replace the input and output amounts with Pedersen Commitments<sup>[def][ecpc~]</sup>. It is then publicly verifiable that the transactions balance (the sum of the committed inputs is greater than the sum of the committed outputs, and all outputs are positive), while keeping the specific committed amounts hidden, thus zero-knowledge. The transaction amounts must be encoded as $ integers \mod q ​$, which can overflow, but is prevented by making use of range proofs. Enter Bulletproofs. The essence of Bulletproofs are its ability to calculate proofs, including range proofs, from inner-products.
```
[the remark this answers is not in the document]
Comment (hansieodendaal) [reply src/cryptography/bulletproofs-and-mimblewimble/MainReport.md]: Both fixed.

File: src/cryptography/bulletproofs-and-mimblewimble/MainReport.md [hunk 4]
```diff
@@ -40,16 +40,29 @@ The essence of Bulletproofs is its inner-product algorithm originally presented
...
+The *prover* must convince the *verifier* that commitment $ C(x,r) = xH + rG $ contains a number such that $ x \in [0,2^n - 1] $. If $ \mathbf {a} = (a_1 \mspace{3mu} , \mspace{3mu} ... \mspace{3mu} , \mspace{3mu} a_n) \in \{0,1\}^n $ is the vector containing the bits of $ x $, the basic idea is to hide all the bits of the amount in a single vector Pedersen Commitment. It must then be proven that each bit satisfies $ \omega(\omega-1) = 0 $, that is each $ \omega $ is either $ 0 $ or $ 1 $, and that they sum to $ x $. As part of the ensuing protocol the *verifier* sends random linear combinations of constraints and challenges $ \in \mathbb{Z_p} $ to the *prover*. The *prover* is then able to construct a vectorized inner product relation containing the elements of $ \mathbf {a} $, the constraints and challenges $ \in \mathbb{Z_p} $ and appropriate blinding vectors $ \in \mathbb Z_p^n ​$.
```
[the remark this answers is not in the document]
Comment (hansieodendaal) [reply src/cryptography/bulletproofs-and-mimblewimble/MainReport.md]: Done

File: src/cryptography/bulletproofs-and-mimblewimble/MainReport.md [hunk 5]
```diff
@@ -40,16 +40,29 @@ The essence of Bulletproofs is its inner-product algorithm originally presented
...
+
+These inner product vectors have size $ n $ that would require many expensive exponentiations. The Pedersen Commitment scheme allows a vector to be cut in half and to compress the two halves together, each time calculating a new set of Pedersen Commitment generators. Applying the same trick repeatedly $ \log _2 n $ times produces a single value. This is applied to the inner product vectors; they are reduced interactively with a logarithmic number of rounds by the *prover* and *verifier* into a single multi-exponentiation of size $ 2n + 2 \log_2(n) + 1 $. This single multi-exponentiation can then be calculated much faster than $ n $ separate ones. All of this are made non-interactive using the Fiat-Shamir Heuristic<sup>[def][fsh~]</sup>.
```
[the remark this answers is not in the document]
Comment (hansieodendaal) [reply src/cryptography/bulletproofs-and-mimblewimble/MainReport.md]: Done

File: src/cryptography/bulletproofs-and-mimblewimble/MainReport.md (hunk 1 above)
Comment (hansieodendaal) [reply src/cryptography/bulletproofs-and-mimblewimble/MainReport.md]: Done

File: src/cryptography/bulletproofs-and-mimblewimble/MainReport.md (hunk 2 above)
Comment (hansieodendaal) [reply src/cryptography/bulletproofs-and-mimblewimble/MainReport.md]: Done

**Pull request merged**